src/Optimization/hiopKKTLinSysSparse.cpp

```diff
@@ -572,70 +602,106 @@ namespace hiop
+          assert(linSys_!=nullptr &&
+                 "HiOp was built without a sparse linear solver for GPU/device and is likely better to "
+                 " run with 'compute_mode' set to 'cpu' instead of the current 'hybrid' "
+                 "(from hiopKKTLinSysCompressedSparseXDYcYd)"); 
+          return nullptr;
+        }
+        // if user requests CPU solvers under hybrid mode, let it run with it he's likely to be aware of the setup
+        assert(linear_solver != "auto");
+        if(nullptr == linSys_ && linear_solver == "ma57") {
 #ifdef HIOP_USE_COINHSL
-        nlp_->log->printf(hovScalars,
-                          "KKT_SPARSE_XDYcYd linsys: alloc MA57 on CPU size %d (%d cons)\n",
-                          n,
-                          neq+nineq);                             
-        linSys_ = new hiopLinSolverIndefSparseMA57(n, nnz, nlp_);
+          nlp_->log->printf(hovScalars,
+                            "KKT_SPARSE_XDYcYd linsys: alloc MA57 on CPU size %d (%d cons) (hybrid)\n",
+                            n,
+                            neq+nineq);                             
+          linSys_ = new hiopLinSolverIndefSparseMA57(n, nnz, nlp_);
 #endif // HIOP_USE_COINHSL
-
-        if(NULL == linSys_) {
+        }
+        
+        if(nullptr == linSys_&& linear_solver == "pardiso") {
 #ifdef HIOP_USE_PARDISO
           nlp_->log->printf(hovScalars,
-                            "KKT_SPARSE_XYcYd linsys: alloc PARDISO on CPU size %d (%d cons)\n",
+                            "KKT_SPARSE_XYcYd linsys: alloc PARDISO on CPU size %d (%d cons) (hybrid)\n",
                             n,
                             neq+nineq);                             
           linSys_ = new hiopLinSolverIndefSparseMA57(n, nnz, nlp_);
-#endif // HIOP_USE_PARDISO
+#endif // HIOP_USE_PARDISO          
         }
-#endif // HIOP_USE_CUSOLVER/STRUMPACK
-      // Add interface to cuSolver/KLU linear solver
-      }
-      assert(linSys_&& "KKT_SPARSE_XDYcYd linsys: cannot instantiate backend linear solver");
+        
+      } else if(compute_mode == "gpu") {
+        assert(false == safe_mode_);
+        /////////////////////////////////////////////////////////////////////////////////////////////
+        // gpu compute mode
+        //
+        // We don't allow CPU linear solvers.
+        /////////////////////////////////////////////////////////////////////////////////////////////
+
+        assert(nullptr == linSys_);
+        if(linear_solver == "cusolver" || linear_solver == "auto") {
+#if defined(HIOP_USE_CUSOLVER)        
+          hiopLinSolverIndefSparseCUSOLVER *p = new hiopLinSolverIndefSparseCUSOLVER(n, nnz, nlp_);
+          nlp_->log->printf(hovScalars,
+                            "KKT_SPARSE_XDYcYd linsys: alloc CUSOLVER size %d (%d cons) (gpu)\n",
+                            n,
+                            neq+nineq);
+          p->setFakeInertia(neq + nineq);
+          linSys_ = p;
```

Comment:
Remove fake inertia:
```suggestion
          hiopLinSolverIndefSparseCUSOLVER *p = new hiopLinSolverIndefSparseCUSOLVER(n, nnz, nlp_);
          nlp_->log->printf(hovScalars,
                            "KKT_SPARSE_XDYcYd linsys: alloc CUSOLVER size %d (%d cons) (gpu)\n",
                            n,
                            neq+nineq);
          linSys_ = p;
```